sys/net/gnrc/network_layer/ndp/node/gnrc_ndp_node.c

```diff
@@ -55,68 +55,66 @@ static gnrc_pktqueue_t *_alloc_pkt_node(gnrc_pktsnip_t *pkt)
     return NULL;
 }
 
-kernel_pid_t gnrc_ndp_node_next_hop_l2addr(uint8_t *l2addr, uint8_t *l2addr_len,
-                                           kernel_pid_t iface, ipv6_addr_t *dst,
-                                           gnrc_pktsnip_t *pkt)
+bool gnrc_ndp_node_next_hop_ipv6_addr(ipv6_addr_t *next_hop_ip,
+                                      kernel_pid_t *iface,
+                                      ipv6_addr_t *dst)
 {
-    gnrc_ipv6_nc_t *nc_entry;
-    ipv6_addr_t *next_hop_ip = NULL, *prefix = NULL;
-
 #ifdef MODULE_GNRC_IPV6_EXT_RH
-    ipv6_hdr_t *hdr;
-    gnrc_pktsnip_t *ipv6;
-    LL_SEARCH_SCALAR(pkt, ipv6, type, GNRC_NETTYPE_IPV6);
-    assert(ipv6);
-    hdr = ipv6->data;
-    next_hop_ip = ipv6_ext_rh_next_hop(hdr);
-#endif
-#ifdef MODULE_FIB
-    ipv6_addr_t next_hop_actual;    /* FIB copies address into this variable */
-    /* don't look-up link local addresses in FIB */
-    if (!ipv6_addr_is_link_local(dst)) {
-        size_t next_hop_size = sizeof(ipv6_addr_t);
-        uint32_t next_hop_flags = 0;
-
-        if ((next_hop_ip == NULL) &&
-            (fib_get_next_hop(gnrc_ipv6_fib_table, &iface, next_hop_actual.u8, &next_hop_size,
-                              &next_hop_flags, (uint8_t *)dst,
-                              sizeof(ipv6_addr_t), 0) >= 0) &&
-            (next_hop_size == sizeof(ipv6_addr_t))) {
-            next_hop_ip = &next_hop_actual;
-        }
+    next_hop_ip = gnrc_ipv6_ext_rh_next_hop(hdr);
```

Comment:
Thanks for the hint. In fact this PR is very outdated and I just left it open to remind me of re-introducing it as a new one anytime soon.